### PR TITLE
Only show user count for large userbases

### PR DESCRIPTION
**Pull request by jacobherrington**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This is a request from one of our early adopter Forems. We should only
render the user count that appears on most of our sign up CTAs when a
Forem has a large userbase (initially, I'm setting that to 1000 users).

The change also removes a lot of duplicated markup by using some
partials to render the copy in the various auth CTAs.

Hopefully, this is written in such a way that changing opening up more
fine-grained control of these partials is trivial.

## Related Tickets & Documents

Closes https://github.com/forem/InternalProjectPlanning/issues/209

## QA Instructions, Screenshots, Recordings

Basically, we drop the number of users until there are more than 1000 users.

Mobile Dropdown Menu:
cta_mobile

Registration Screen:
cta_registration

In-feed CTA:
cta_feed

Sidebar:
cta_sidebar

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

**Review by juliannatetreault — APPROVED** [review]
I love the new partial! 😄  Everything showed as expected both on desktop and mobile in all locations that the copy was altered. 👍 👍

**Review by rhymes — COMMENTED** [review]
Cool idea! Though small communities might still be interested in showing the number of members.

File: app/views/shared/_authentication_title.html.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+<a href="/"><%= community_name %></a> is a community of <%= display_estimated_user_count? ? number_with_delimiter(estimated_user_count) : "" %> amazing <%= community_members_label %>
```
Comment (ludwiczakpawel) [inline app/views/shared/_authentication_title.html.erb:1]: Just wondering... Should we start using  `<%= root_path %>` instead of `/` in href attribute?

**Review by ludwiczakpawel — APPROVED** [review]
👌

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -11,6 +11,8 @@ module ApplicationHelper
   )
   # rubocop:enable Performance/OpenStruct
 
+  LARGE_USERBASE_THRESHOLD = 1000
```
[the remark this answers is not in the document]
Comment (jacobherrington) [reply app/helpers/application_helper.rb:14]: I put it in a constant because we might add some knobs here, but decided not to do so yet 😁

**Comment by jacobherrington** [comment]
@rhymes I agree, this should be configurable in the near future. This is a compromise we settled on to satisfy a request from Snyk. That's why I made it super easy to turn into a SiteConfig or something soon 😁
